File: src/main/java/br/com/unopay/api/market/repository/BonusBillingRepository.java
package br.com.unopay.api.market.repository;

import br.com.unopay.api.market.model.BonusBilling;
import br.com.unopay.api.market.model.filter.BonusBillingFilter;
import br.com.unopay.bootcommons.repository.filter.UnovationFilterRepository;

import java.util.Optional;

public interface BonusBillingRepository extends UnovationFilterRepository<BonusBilling, String, BonusBillingFilter> {
    Optional<BonusBilling> findById(String id);
}

File: src/main/scala/br/com/unopay/api/market/repository/BonusBillingRepository.scala
package br.com.unopay.api.market.repository

import br.com.unopay.api.market.model.BonusBilling
import br.com.unopay.api.market.model.filter.BonusBillingFilter
import br.com.unopay.bootcommons.repository.filter.UnovationFilterRepository

import java.util.Optional

trait BonusBillingRepository extends UnovationFilterRepository[BonusBilling, String, BonusBillingFilter] {

    def findById(id: String): Optional[BonusBilling]
    def findByIdAndPayerId(id: String, payerId: String): Optional[BonusBilling]
    def findFirstByOrderByCreatedDateTimeDesc(): Optional[BonusBilling]
}
